xbmc/FileItem.cpp

```diff
@@ -3533,17 +3534,42 @@ int CFileItem::GetVideoContentType() const
   return type;
 }
 
+CFileItem CFileItem::GetItemToPlay() const
+{
+  if (HasEPGInfoTag())
+  {
+    const CPVRChannelPtr channel(GetEPGInfoTag()->ChannelTag());
+    if (channel)
+      return CFileItem(channel);
+  }
+  return CFileItem(*this);
```

Comment:
This would require to derive CFileItem from std::enable_shared_from_this which I wanted to prevent.